platform/security/certificate/src/main/java/org/codice/ddf/certificate/CertificateManager.java

```diff
@@ -0,0 +1,267 @@
+import java.util.Date;
+
+import org.bouncycastle.asn1.ASN1ObjectIdentifier;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x509.AlgorithmIdentifier;
+import org.bouncycastle.asn1.x509.SubjectPublicKeyInfo;
+import org.bouncycastle.cert.X509v3CertificateBuilder;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.operator.ContentSigner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+    /*+----------------------------------------------------------------------
+     ||
+     ||  Class: CertificateManager
+     ||
+     ||        Purpose:  Collection of static methods that manages X509Certificate creation and signing.
+     ||
+     ||  Inherits From:  None
+     ||
+     ||     Interfaces:  None
+     ||
+     |+-----------------------------------------------------------------------
+     ||
+     ||      Constants:  MILLIS_IN_DAY - Number of milliseconds in a day
+     ||                  MILLIS_IN_YEAR - Number of milliseconds in a year
+     ||
+     |+-----------------------------------------------------------------------
+     ||
+     ||  Class Methods:     X509v3CertificateBuilder    createDefaultX509SigningRequest(PublicKey)
+     ||                     X509v3CertificateBuilder    createCustomX509SigningRequest(String, String, int, PublicKey)
+     ||                     X509v3CertificateBuilder    createCustomX509SigningRequest(String, String, String, String, String, String, String, String, String, String, int, PublicKey)
+     ||                     X509Certificate             signX509CertificateSHA256RSA(X509v3CertificateBuilder, PrivateKey)
+     ||                     void                        registerBCSecurityProvider()
+     ||                     String                      getHostName()
+     ||
+     ||
+     ++-----------------------------------------------------------------------*/
+
+public class CertificateManager {
+    private static final long MILLIS_IN_DAY = 86400000;
+
+    private static final long MILLIS_IN_YEAR = 31536000000L;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CertificateManager.class);
+
+    /**
+     * ---------------------------------------------------------------------
+     * Method: createDefaultX509SigningRequest
+     * <p/>
+     * Purpose:  Creates a signing request with default issuer and subject information
+     * <p/>
+     * Pre-condition: None
+     * <p/>
+     * Post-condition: Issuer information, subject information, and the encoded public key have been added to the returned signing request. Valid for 3 years.
+     *
+     * @param pubKey the PublicKey to be associated with the certificate signing request
+     * @return X509v3CertificateBuilder with the issuer, subject, and encoded public key set.
+     * -------------------------------------------------------------------
+     */
+
+    public static X509v3CertificateBuilder createDefaultX509SigningRequest(PublicKey pubKey) {
```

Comment:
Any reason these methods are static? Makes them really hard to mock when testing other classes.